hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/defaultserializers/CopyOnWriteArrayListStreamSerializer.java

```diff
@@ -42,4 +45,18 @@ public int getTypeId() {
 
         return new CopyOnWriteArrayList<>(collection);
     }
+
+    @Override
+    @SuppressWarnings("DuplicatedCode")
+    public void write(ObjectDataOutput out, CopyOnWriteArrayList<E> collection) throws IOException {
+        Spliterator<E> cowSplitIterator = collection.spliterator();
+        out.writeInt((int) cowSplitIterator.getExactSizeIfKnown());
```

Comment:
`getExactSizeIfKnown` can return -1 according to the contract. I guess it won't for COWAL, but we can assert that.

Reply:
Done https://github.com/hazelcast/hazelcast/pull/18285/commits/4e1a00c4b0aa682df12c7bc2ec8d252df9acbb00

Reply:
Contract of CopyOnWriteArrayList states that this never return -1. 
So this behaviour can not ever change.